mesh-design.md

```diff
@@ -0,0 +1,270 @@
+		name: string, height: number, radius: number,
+		direction: 'x' | 'y' | 'z' = 'y', uSegments = 36
+	): Mesh { }
+	public createPlaneMesh(name: string, width: number, height: number, uSegments = 1, vSegments = 1): Mesh { }
+
+	// generic form
+	public createPrimitiveMesh(name: string, definition: PrimitiveDefinition): Mesh { }
+}
+```
+
+Or loaded from `AssetContainer.loadGltf`, and returned with that list of assets.
+
+## Usage
+
+A `meshId` property would be added to `AppearanceLike`, and a corresponding `mesh` property to `Appearance`.
+
+Assigning a new primitive mesh to an existing actor:
+
+```ts
+const assets: MRE.AssetContainer;
+const actor: MRE.Actor;
+actor.appearance.mesh = assets.createBoxPrimitive('box', 1, 1, 1);
+```
+
+Creating a new actor with a new primitive mesh:
+
+```ts
+const context: MRE.Context;
+const assets: MRE.AssetContainer;
+MRE.Actor.CreateEmpty(context, {
+	name: 'box',
+	appearance: {
+		meshId: assets.createBoxPrimitive('box', 1, 1, 1).id
+	}
+});
+```
+
+Using a mesh from a glTF:
+
+```ts
+const assets: MRE.AssetContainer;
+const actor: MRE.Actor;
+await assets.loadGltf('myMesh.gltf');
+actor.appearance.mesh = assets.meshes.find(m => m.name === "Asset I want") || assets.meshes[0];
+```
+
+## Colliders
+
+Today, colliders are created in one of three ways. On actor creation:
+
+1. Through the `addCollider` option of `Actor.CreatePrimitive`, which takes a boolean. The shape and size of
+	the resulting collider is determined by the primitive definition. The collider is built Unity-side, then
+	its description is returned in the generated actor patch.
+2. Through the `colliderType` option of `Actor.CreateFromGltf`, which accepts any of our known collider types:
+	`box`, `sphere`, `mesh`, `capsule`, or `none`. Of these, UnityGLTF only knows how to process `box` and `mesh`;
+	the others will be ignored. A separate collider will be generated for each mesh in the glTF. For `box` types,
+	the size is determined by the axis-aligned bounding box of the mesh. The colliders are built Unity-side, then
+	their descriptions are returned in the generated actor patches.
+
+Or at runtime:
+
+3. Through an Actor instance's `setCollider(colliderType: 'box' | 'sphere', ...moreOptions)` method, which takes
+	the shape, size, and options of the desired collider, and adds it to the accumulating actor patch. Next
+	`actor-update`, that collider patch is converted into a Unity collider of the appropriate type.
+
+**Note**: Mesh colliders in Unity are [expensive to generate, expensive to evaluate, and have a laundry list of
+caveats](https://docs.unity3d.com/Manual/class-MeshCollider.html). For this reason they generally advise against
+adding mesh colliders at all at runtime. So if we want MREs to have the ability to produce mesh colliders at all,
+we should be prepared for those performance penalties. For this reason, only `CreateFromGltf` can generate mesh
+colliders, though I suspect this isn't great either.
```

Comment:
Let's do what we talked about at lunch and allow 'mesh' colliders to be generated everywhere and just document that it's costly. The fact that it's not performant is a Unity limitation, but may not apply to future hosts (Babylon, etc).

Reply:
I do hope lunch was tasty and filled with an enjoyable conversation about mesh colliders.  Sorry I couldn't attend the meeting due to my location being in NJ.  What was the conversation about and how did we come to this conclusion?

Reply:
Apologies Tom! That was insensitive of me. That part of my comment was completely unnecessary, and could have simply read:
> Let's allow 'mesh' colliders to be generated everywhere and just document that it's costly. The fact that it's not performant is a Unity limitation, but may not apply to future hosts (Babylon, etc).

But I think I didn't fully grok the alternative approach that exposes 'auto'. I didn't realize it removed 'mesh' entirely. I think that's a better, simpler solution. I retract my earlier recommendation of exposing 'mesh' everywhere in favor of 'auto'.